test/objects.js

```diff
@@ -609,8 +609,12 @@
       var view2 = new Uint8Array(shared.buffer, 2, 2);
       assert.notOk(_.isEqual(view1, view2), 'same buffer with different offset is not equal');
 
-      // Some older browsers support typed arrays but not DataView.
-      if (typeof DataView !== 'undefined') {
+      if (typeof NativeDataView !== 'undefined') { // Still test overriden DataView
+        assert.ok(_.isEqual(new NativeDataView(u8.buffer), new NativeDataView(u8b.buffer)), 'Identical NativeDataView are equal');
+        assert.ok(_.isEqual(new NativeDataView(u8.buffer), new NativeDataView(i8.buffer)), 'Identical NativeDataView of different typed arrays are equal');
+        assert.notOk(_.isEqual(new NativeDataView(u8.buffer), new NativeDataView(u16.buffer)), 'Different NativeDataView with different length are not equal');
+        assert.notOk(_.isEqual(new NativeDataView(u8.buffer), new NativeDataView(u16one.buffer)), 'Different NativeDataView with different byte data are not equal');
```

Comment:
These seem to be copies of the lines that come after the `else`, in which you only replaced `DataView` by `NativeDataView`. It would be elegant if you could avoid the duplicate code. Something like this:

```js
var DataViewCtor = (
  typeof NativeDataView !== 'undefined' ? NativeDataView :
  typeof DataView !== 'undefined' ? DataView :
  undefined
);
if (DataViewCtor) {
  assert.ok(_.isEqual(new DataViewCtor(u8.buffer), ...));
  // etcetera
}
```

Reply:
I had actually thought about this at one point and then must have gotten distracted and forgotten to clean it up, thanks for the reminder, fixed